src/main/kotlin/io/libp2p/core/protocol/Negotiator.kt

```diff
@@ -37,20 +38,34 @@ object Negotiator {
     private const val TIMEOUT_MILLIS: Long = 10_000
     private const val MULTISTREAM_PROTO = "/multistream/1.0.0"
 
-    private val MULTISTREAM_PROTO_BYTES = MULTISTREAM_PROTO.toByteArray().toByteBuf()
-    private val NEWLINE = "\n".toByteArray().toByteBuf()
-    private val NA = "na".toByteArray().toByteBuf()
+    private val HEADING = '\u0013'
```

Comment:
Where does this heading come from? I think it might be a uvarint length prefix?

Reply:
Was also a bit surprised. Do you have specs on this? Do we need len prefix for line based protocol?